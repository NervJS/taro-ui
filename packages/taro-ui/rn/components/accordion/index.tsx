import classNames from 'classnames'
import PropTypes, { InferProps } from 'prop-types'
import React from 'react'
import { Text, View } from '@tarojs/components'
import { CommonEvent } from '@tarojs/components/types/common'
import { AtAccordionProps, AtAccordionState } from '../../../types/accordion'
import { delayQuerySelector } from '../../common/utils'

export default class AtAccordion extends React.Component<
  AtAccordionProps,
  AtAccordionState
> {
  private isCompleted: boolean
  private startOpen: boolean

  public static defaultProps: AtAccordionProps
  public static propTypes: InferProps<AtAccordionProps>

  public constructor(props: AtAccordionProps) {
    super(props)
    this.isCompleted = true
    this.startOpen = false
    this.state = {
      wrapperHeight: 0
    }
  }

  private handleClick = (event: CommonEvent): void => {
    const { open } = this.props
    if (!this.isCompleted) return

    this.props.onClick && this.props.onClick(!open, event)
  }

  private toggleWithAnimation(): void {
    const { open, isAnimation } = this.props
    if (!this.isCompleted || !isAnimation) return

    this.isCompleted = false
    delayQuerySelector('.at-accordion__body', 0).then(rect => {
      const height = parseInt(rect[0].height.toString())
      const startHeight = open ? height : 0
      const endHeight = open ? 0 : height
      this.startOpen = false
      this.setState(
        {
          wrapperHeight: startHeight
        },
        () => {
          setTimeout(() => {
            this.setState(
              {
                wrapperHeight: endHeight
              },
              () => {
                setTimeout(() => {
                  this.isCompleted = true
                  this.setState({})
                }, 700)
              }
            )
          }, 100)
        }
      )
    })
  }

  public UNSAFE_componentWillReceiveProps(nextProps: AtAccordionProps): void {
    if (nextProps.open !== this.props.open) {
      this.startOpen = !!nextProps.open && !!nextProps.isAnimation
      this.toggleWithAnimation()
    }
  }

  public render(): JSX.Element {
    const {
      customStyle,
      className,
      title,
      icon,
      hasBorder,
      open,
      note
    } = this.props
    const { wrapperHeight } = this.state

    const rootCls = classNames('at-accordion', className)
    const prefixClass = (icon && icon.prefixClass) || 'at-icon'
    const iconCls = classNames({
      [prefixClass]: true,
      [`${prefixClass}-${icon && icon.value}`]: icon && icon.value,
      'at-accordion__icon': true
    })
    const headerCls = classNames('at-accordion__header', {
      'at-accordion__header--noborder': !hasBorder
    })
    const arrowCls = classNames('at-accordion__arrow', {
      'at-accordion__arrow--folded': !!open
    })
    const contentCls = classNames('at-accordion__content', {
      'at-accordion__content--inactive':
        (!open && this.isCompleted) || this.startOpen
    })
    const iconStyle = {
      color: (icon && icon.color) || '',
      fontSize: (icon && `${icon.size}px`) || ''
    }
    const contentStyle = { height: `${wrapperHeight}px` }

    if (this.isCompleted) {
      contentStyle.height = ''
    }

    return (
      <View className={rootCls} style={customStyle}>
        <View className={headerCls} onClick={this.handleClick}>
          {icon && icon.value && (
            <Text className={iconCls} style={iconStyle}></Text>
          )}
          <View className='at-accordion__info'>
            <View className='at-accordion__info__title'>{title}</View>
            <View className='at-accordion__info__note'>{note}</View>
          </View>
          <View className={arrowCls}>
            <Text className='at-icon at-icon-chevron-down'></Text>
          </View>
        </View>
        <View style={contentStyle} className={contentCls}>
          <View className='at-accordion__body'>{this.props.children}</View>
        </View>
      </View>
    )
  }
}

AtAccordion.defaultProps = {
  open: false,
  customStyle: '',
  className: '',
  title: '',
  note: '',
  icon: { value: '' },
  hasBorder: true,
  isAnimation: true
}

AtAccordion.propTypes = {
  customStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.string]),
  className: PropTypes.oneOfType([PropTypes.array, PropTypes.string]),
  open: PropTypes.bool,
  isAnimation: PropTypes.bool,
  title: PropTypes.string,
  note: PropTypes.string,
  icon: PropTypes.object,
  hasBorder: PropTypes.bool,
  onClick: PropTypes.func
}
