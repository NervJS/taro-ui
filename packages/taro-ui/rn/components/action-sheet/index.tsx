import classNames from 'classnames'
import PropTypes, { InferProps } from 'prop-types'
import React from 'react'
import { View } from '@tarojs/components'
import Modal from 'react-native-modal'
import { CommonEvent } from '@tarojs/components/types/common'
import {
  AtActionSheetProps,
  AtActionSheetState,
} from '../../../types/action-sheet'
import AtActionSheetBody from './body/index'
import AtActionSheetFooter from './footer/index'
import AtActionSheetHeader from './header/index'

export default class AtActionSheet extends React.Component<
  AtActionSheetProps,
  AtActionSheetState
> {
  public static defaultProps: AtActionSheetProps
  public static propTypes: InferProps<AtActionSheetProps>

  public constructor(props: AtActionSheetProps) {
    super(props)
    const { isOpened } = props

    this.state = {
      _isOpened: isOpened,
    }
  }

  public UNSAFE_componentWillReceiveProps(nextProps: AtActionSheetProps): void {
    const { isOpened } = nextProps
    if (isOpened !== this.state._isOpened) {
      this.setState({
        _isOpened: isOpened,
      })

      !isOpened && this.handleClose()
    }
  }

  private handleClose = (): void => {
    if (typeof this.props.onClose === 'function') {
      this.props.onClose()
    }
  }

  private handleCancel = (): void => {
    if (typeof this.props.onCancel === 'function') {
      return this.props.onCancel()
    }
    this.close()
  }

  private close = (): void => {
    this.setState(
      {
        _isOpened: false,
      },
      this.handleClose,
    )
  }

  private handleTouchMove = (e: CommonEvent): void => {
    e.stopPropagation()
    e.preventDefault()
  }

  public render(): JSX.Element {
    const { title, cancelText, className } = this.props
    const { _isOpened } = this.state

    // const rootClass = classNames(
    //   'at-action-sheet',
    //   {
    //     'at-action-sheet--active': _isOpened
    //   },
    //   className
    // )
    const containerClass = classNames('at-action-sheet__container', className)

    return (
      <Modal
        animationIn='fadeInUp'
        animationOut='fadeOutDown'
        isVisible={_isOpened}
        hasBackdrop
        onModalHide={this.handleClose}
        onBackButtonPress={this.handleClose}
        onBackdropPress={this.handleClose}
        style={{
          margin: 0,
        }}
      >
        <View className={containerClass}>
          {title && <AtActionSheetHeader>{title}</AtActionSheetHeader>}
          <AtActionSheetBody>{this.props.children}</AtActionSheetBody>
          {cancelText && (
            <AtActionSheetFooter onClick={this.handleCancel}>
              {cancelText}
            </AtActionSheetFooter>
          )}
        </View>
      </Modal>
    )
  }
}

AtActionSheet.defaultProps = {
  title: '',
  cancelText: '',
  isOpened: false,
}

AtActionSheet.propTypes = {
  title: PropTypes.string,
  onClose: PropTypes.func,
  onCancel: PropTypes.func,
  isOpened: PropTypes.bool.isRequired,
  cancelText: PropTypes.string,
}
