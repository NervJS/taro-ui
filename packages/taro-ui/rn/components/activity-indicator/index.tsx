import classNames from 'classnames'
import PropTypes, { InferProps } from 'prop-types'
import React from 'react'
import { Text, View } from '@tarojs/components'
import { AtActivityIndicatorProps } from '../../../types/activity-indicator'
import AtLoading from '../loading/index'

export default class AtActivityIndicator extends React.Component<
  AtActivityIndicatorProps
> {
  public static defaultProps: AtActivityIndicatorProps
  public static propTypes: InferProps<AtActivityIndicatorProps>

  public render(): JSX.Element {
    const { color, size, mode, content, isOpened } = this.props

    const rootClass = classNames(
      'at-activity-indicator',
      {
        'at-activity-indicator--center': mode === 'center',
        'at-activity-indicator--isopened': isOpened
      },
      this.props.className
    )

    return (
      <View className={rootClass}>
        <View className='at-activity-indicator__body'>
          <AtLoading size={size} color={color} />
        </View>
        {content && (
          <Text className='at-activity-indicator__content'>{content}</Text>
        )}
      </View>
    )
  }
}

AtActivityIndicator.defaultProps = {
  size: 0,
  mode: 'normal',
  color: '',
  content: '',
  className: '',
  isOpened: true
}

AtActivityIndicator.propTypes = {
  size: PropTypes.number,
  mode: PropTypes.string,
  color: PropTypes.string,
  content: PropTypes.string,
  className: PropTypes.oneOfType([PropTypes.array, PropTypes.string]),
  isOpened: PropTypes.bool
}
