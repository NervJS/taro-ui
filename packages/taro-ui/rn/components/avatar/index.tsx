import classNames from 'classnames'
import PropTypes, { InferProps } from 'prop-types'
import React from 'react'
import { Image, View } from '@tarojs/components'
import Taro from '@tarojs/taro'
import { AtAvatarProps, AtAvatarState } from '../../../types/avatar'

let OpenData: any = null
if (process.env.TARO_ENV === 'rn') {
  OpenData = View
} else {
  OpenData = require('@tarojs/components')
}

const SIZE_CLASS = {
  large: 'large',
  normal: 'normal',
  small: 'small',
}

export default class AtAvatar extends React.Component<
  AtAvatarProps,
  AtAvatarState
> {
  public static defaultProps: AtAvatarProps
  public static propTypes: InferProps<AtAvatarProps>

  public constructor(props: AtAvatarProps) {
    super(props)
    this.state = {
      isWEAPP: Taro.getEnv() === Taro.ENV_TYPE.WEAPP,
      _width: 0,
    }
  }

  private onLayout = (event: any): void => {
    const { width } = event.nativeEvent.layout
    this.setState({ _width: width })
  }

  public render(): JSX.Element {
    const { _width } = this.state
    const { size, circle, image, text, openData, customStyle } = this.props
    const rootClassName = ['at-avatar']
    const iconSize = SIZE_CLASS[size || 'normal']
    const classObject = {
      [`at-avatar--${iconSize}`]: iconSize,
      'at-avatar--circle': circle,
    }

    let letter = ''
    if (text) letter = text[0]

    let elem: React.ReactNode
    if (openData && openData.type === 'userAvatarUrl' && this.state.isWEAPP) {
      elem = <OpenData type={openData.type}></OpenData>
    } else if (image) {
      elem = <Image className='at-avatar__img' src={image} />
    } else {
      // elem = <Text className='at-avatar__text'>{letter}</Text>
      // TODO: RN
      elem = letter
    }

    let style = customStyle
    if (typeof customStyle === 'object' && _width && circle) {
      style = { ...customStyle, borderRadius: _width / 2 }
    }

    return (
      <View
        className={classNames(rootClassName, classObject, this.props.className)}
        style={style}
        onLayout={this.onLayout}
      >
        {elem}
      </View>
    )
  }
}

AtAvatar.defaultProps = {
  size: 'normal',
  circle: false,
  text: '',
  image: '',
  customStyle: {},
  className: '',
}

AtAvatar.propTypes = {
  size: PropTypes.oneOf(['large', 'normal', 'small']),
  circle: PropTypes.bool,
  text: PropTypes.string,
  image: PropTypes.string,
  openData: PropTypes.object,
  customStyle: PropTypes.oneOfType([PropTypes.object]),
  className: PropTypes.oneOfType([PropTypes.array, PropTypes.string]),
}
