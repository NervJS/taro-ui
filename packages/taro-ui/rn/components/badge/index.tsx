import classNames from 'classnames'
import PropTypes, { InferProps } from 'prop-types'
import React from 'react'
import { View } from '@tarojs/components'
import { AtBadgeProps } from '../../../types/badge'

export default class AtBadge extends React.Component<AtBadgeProps> {
  public static defaultProps: AtBadgeProps
  public static propTypes: InferProps<AtBadgeProps>

  public constructor(props: AtBadgeProps) {
    super(props)
    this.state = {}
  }

  private formatValue(
    value: string | number | undefined,
    maxValue: number
  ): string | number {
    if (value === '' || value === null || typeof value === 'undefined')
      return ''
    const numValue = +value
    if (Number.isNaN(numValue)) {
      return value
    }
    return numValue > maxValue ? `${maxValue}+` : numValue
  }

  public render(): JSX.Element {
    const { dot, value, maxValue = 99, customStyle } = this.props
    const rootClassName = ['at-badge']

    const val = this.formatValue(value, maxValue)

    return (
      <View
        className={classNames(rootClassName, this.props.className)}
        style={customStyle}
      >
        {this.props.children}
        {dot ? (
          <View className='at-badge__dot'></View>
        ) : (
          val !== '' && <View className='at-badge__num'>{val}</View>
        )}
      </View>
    )
  }
}

AtBadge.defaultProps = {
  dot: false,
  value: '',
  maxValue: 99,
  customStyle: {},
  className: ''
}

AtBadge.propTypes = {
  dot: PropTypes.bool,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  maxValue: PropTypes.number,
  customStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.string]),
  className: PropTypes.oneOfType([PropTypes.array, PropTypes.string])
}
