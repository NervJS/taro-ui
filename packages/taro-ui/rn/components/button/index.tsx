/* eslint-disable @typescript-eslint/ban-ts-ignore */
import classNames from 'classnames'
import PropTypes, { InferProps } from 'prop-types'
import React from 'react'
import { Button, Text } from '@tarojs/components'
import { CommonEvent } from '@tarojs/components/types/common'
import Taro from '@tarojs/taro'
import { AtButtonProps, AtButtonState } from '../../../types/button'

const SIZE_CLASS = {
  normal: 'normal',
  small: 'small'
}

const TYPE_CLASS = {
  primary: 'primary',
  secondary: 'secondary'
}

export default class AtButton extends React.Component<
  AtButtonProps,
  AtButtonState
> {
  public static defaultProps: AtButtonProps
  public static propTypes: InferProps<AtButtonProps>

  public constructor(props: AtButtonProps) {
    super(props)
    this.state = {
      isWEB: Taro.getEnv() === Taro.ENV_TYPE.WEB,
      isWEAPP: Taro.getEnv() === Taro.ENV_TYPE.WEAPP,
      isALIPAY: Taro.getEnv() === Taro.ENV_TYPE.ALIPAY
    }
  }

  private onClick(event: CommonEvent): void {
    this.props.onClick && this.props.onClick(event)
  }

  public render(): JSX.Element {
    const {
      size = 'normal',
      type = '',
      circle,
      full,
      loading,
      disabled,
      customStyle
    } = this.props
    const rootClassName = ['at-button']
    const classObject = {
      'at-button--disabled': disabled,
      [`at-button--${type}`]: TYPE_CLASS[type],
      [`at-button--${SIZE_CLASS[size]}`]: SIZE_CLASS[size],
      'at-button--circle': circle,
      'at-button--full': full
    }

    return (
      <Button
        disabled={disabled}
        className={classNames(rootClassName, classObject, this.props.className)}
        style={customStyle}
        // @ts-ignore
        hoverStyle={{
          opacity: 0.6
        }}
        loading={loading}
        type={type !== 'primary' ? 'default' : 'primary'}
        size={size === 'normal' ? 'default' : 'mini'}
        onClick={this.onClick.bind(this)}
      >
        {type === 'secondary' && this.props.children ? (
          <Text
            className={`at-button--secondary--text at-button--${size}--text`}
          >
            {this.props.children}
          </Text>
        ) : (
          this.props.children
        )}
      </Button>
    )
  }
}

AtButton.defaultProps = {
  size: 'normal',
  circle: false,
  full: false,
  loading: false,
  disabled: false,
  customStyle: {},
  // Button props
  lang: 'en',
  sessionFrom: '',
  sendMessageTitle: '',
  sendMessagePath: '',
  sendMessageImg: '',
  showMessageCard: false,
  appParameter: ''
}

AtButton.propTypes = {
  size: PropTypes.oneOf(['normal', 'small']),
  type: PropTypes.oneOf(['primary', 'secondary', '']),
  circle: PropTypes.bool,
  full: PropTypes.bool,
  loading: PropTypes.bool,
  disabled: PropTypes.bool,
  onClick: PropTypes.func,
  customStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.string]),
  formType: PropTypes.oneOf(['submit', 'reset', '']),
  openType: PropTypes.oneOf([
    'contact',
    'share',
    'getUserInfo',
    'getPhoneNumber',
    'launchApp',
    'openSetting',
    'feedback',
    'getRealnameAuthInfo',
    'getAuthorize',
    'contactShare',
    ''
  ]),
  lang: PropTypes.string,
  sessionFrom: PropTypes.string,
  sendMessageTitle: PropTypes.string,
  sendMessagePath: PropTypes.string,
  sendMessageImg: PropTypes.string,
  showMessageCard: PropTypes.bool,
  appParameter: PropTypes.string,
  onGetUserInfo: PropTypes.func,
  onContact: PropTypes.func,
  onGetPhoneNumber: PropTypes.func,
  onError: PropTypes.func,
  onOpenSetting: PropTypes.func
}
