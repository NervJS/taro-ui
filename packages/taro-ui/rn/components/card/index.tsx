import classNames from 'classnames'
import PropTypes, { InferProps } from 'prop-types'
import React from 'react'
import { Image, Text, View } from '@tarojs/components'
import AtIcon from '../icon'
import { AtCardProps } from '../../../types/card'

export default class AtCard extends React.Component<AtCardProps> {
  public static defaultProps: AtCardProps
  public static propTypes: InferProps<AtCardProps>

  private handleClick = (args: any): void => {
    if (typeof this.props.onClick === 'function') {
      this.props.onClick(args)
    }
  }

  public render(): JSX.Element {
    const { title, note, extra, extraStyle, thumb, isFull, icon, renderIcon } =
      this.props

    const rootClass = classNames(
      'at-card',
      {
        'at-card--full': isFull,
      },
      this.props.className,
    )
    // const iconClass = classNames({
    //   'at-icon': true,
    //   [`at-icon-${icon && icon.value}`]: icon && icon.value,
    //   'at-card__header-icon': true
    // })

    // const iconStyle = {
    //   color: (icon && icon.color) || '',
    //   fontSize: (icon && `${icon.size}px`) || ''
    // }

    return (
      <View onClick={this.handleClick} className={rootClass}>
        <View className='at-card__header'>
          {thumb && (
            <View className='at-card__header-thumb'>
              <Image
                className='at-card__header-thumb-info'
                mode='scaleToFill'
                src={thumb}
              />
            </View>
          )}
          {renderIcon || ''}
          {!thumb && icon && icon.value && (
            <AtIcon
              customStyle={icon.customStyle}
              value={icon.value}
              size={icon.size}
              color={icon.color}
            />
          )}

          <Text className='at-card__header-title'>{title}</Text>
          {extra && (
            <Text style={{ ...extraStyle }} className='at-card__header-extra'>
              {extra}
            </Text>
          )}
        </View>
        <View className='at-card__content'>
          <View className='at-card__content-info'>{this.props.children}</View>
          {note ? <View className='at-card__content-note'>{note}</View> : null}
        </View>
      </View>
    )
  }
}

AtCard.defaultProps = {
  note: '',
  isFull: false,
  thumb: '',
  title: '',
  extra: '',
  extraStyle: {},
}

AtCard.propTypes = {
  note: PropTypes.string,
  isFull: PropTypes.bool,
  thumb: PropTypes.string,
  title: PropTypes.string,
  extra: PropTypes.string,
  icon: PropTypes.object,
  onClick: PropTypes.func,
  renderIcon: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
  extraStyle: PropTypes.object, // 自定义extra样式
}
