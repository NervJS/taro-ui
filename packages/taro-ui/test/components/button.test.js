import React from 'react'
import { mount } from 'enzyme'
import { Button } from '@tarojs/components'
import AtButton from '../../lib/components/button'
import AtLoading from '../../lib/components/loading'

const colorStyle = {
  border: '1px solid #fff',
  borderColor: '#fff transparent transparent transparent'
}

const sizeStyle = {
  width: '30rpx', // TODO: 应该是 rem
  height: '30rpx'
}

describe('Button Component', () => {
  it('render initial Button', () => {
    const component = mount(<AtButton>按钮</AtButton>)
    expect(component.find('taro-view-core')).toHaveLength(2)
    expect(component.find('taro-button-core')).toHaveLength(1)
    expect(component.find('.at-button').length).toBeGreaterThanOrEqual(1)
    expect(
      component.find('.at-button__wxbutton').length
    ).toBeGreaterThanOrEqual(1)
    expect(component.find('.at-button__text').first().text()).toBe('按钮')
    expect(component.html()).toMatchSnapshot()
  })

  it('render Button -- props type(primary)', () => {
    const component = mount(<AtButton type='primary'>按钮</AtButton>)
    expect(component.prop('type')).toBe('primary')
    expect(component.find('.at-button--primary').length).toBeGreaterThanOrEqual(
      1
    )
    expect(component.html()).toMatchSnapshot()
  })

  it('render Button -- props type(secondary)', () => {
    const component = mount(<AtButton type='secondary'>按钮</AtButton>)
    expect(component.prop('type')).toBe('secondary')
    expect(
      component.find('.at-button--secondary').length
    ).toBeGreaterThanOrEqual(1)
    expect(component.html()).toMatchSnapshot()
  })

  it('render Button -- props size(normal)', () => {
    const component = mount(<AtButton size='normal'>按钮</AtButton>)
    expect(component.prop('size')).toBe('normal')
    expect(component.find('.at-button--normal').length).toBeGreaterThanOrEqual(
      1
    )
    expect(component.html()).toMatchSnapshot()
  })

  it('render Button -- props size(small)', () => {
    const component = mount(<AtButton size='small'>按钮</AtButton>)
    expect(component.prop('size')).toBe('small')
    expect(component.find('.at-button--small').length).toBeGreaterThanOrEqual(1)
    expect(component.html()).toMatchSnapshot()
  })

  it('render Button -- props circle', () => {
    const component = mount(<AtButton circle>按钮</AtButton>)
    expect(component.find('.at-button--circle').length).toBeGreaterThanOrEqual(
      1
    )
    expect(component.html()).toMatchSnapshot()
  })

  it('render Button -- props full', () => {
    const component = mount(<AtButton full>按钮</AtButton>)
    expect(component.find('.at-button--full').length).toBeGreaterThanOrEqual(1)
    expect(component.html()).toMatchSnapshot()
  })

  it('render Button -- props loading', () => {
    const component = mount(<AtButton loading>按钮</AtButton>)
    expect(component.find('taro-view-core')).toHaveLength(7)
    expect(component.find(AtLoading)).toHaveLength(1)
    expect(
      component.find('.at-button.at-button--icon').length
    ).toBeGreaterThanOrEqual(1)
    expect(component.html()).toMatchSnapshot()
  })

  it('render Button -- props loading(with primary type)', () => {
    const component = mount(
      <AtButton loading type='primary'>
        按钮
      </AtButton>
    )
    expect(component.prop('type')).toBe('primary')
    expect(component.find(AtLoading).prop('color')).toBe('#fff')
    expect(
      component.find('.at-loading__ring').first().prop('style')
    ).toMatchObject(colorStyle)
    expect(component.html()).toMatchSnapshot()
  })

  it('render Button -- props loading(with small size)', () => {
    const component = mount(
      <AtButton loading size='small'>
        按钮
      </AtButton>
    )
    expect(component.prop('size')).toBe('small')
    expect(component.find(AtLoading).prop('size')).toBe('30')
    expect(
      component.find('.at-loading__ring').first().prop('style')
    ).toMatchObject(sizeStyle)
    expect(component.html()).toMatchSnapshot()
  })

  it('render Button -- props disabled', () => {
    const component = mount(<AtButton disabled>按钮</AtButton>)
    expect(
      component.find('.at-button--disabled').length
    ).toBeGreaterThanOrEqual(1)
    expect(component.html()).toMatchSnapshot()
  })

  it('render Button -- props formType', () => {
    const component = mount(<AtButton formType='submit'>按钮</AtButton>)
    expect(component.prop('formType')).toBe('submit')
    expect(component.find(Button).prop('formType')).toBe('submit')
    expect(component.html()).toMatchSnapshot()
  })

  it('render Button -- props lang', () => {
    const component = mount(<AtButton lang='zh_CN'>按钮</AtButton>)
    expect(component.prop('lang')).toBe('zh_CN')
    expect(component.find(Button).prop('lang')).toBe('zh_CN')
    expect(component.html()).toMatchSnapshot()
  })
})

describe('Button Behavior', () => {
  it('handle Button -- event onClick', () => {
    const handleClick = jest.fn()
    const component = mount(<AtButton onClick={handleClick}>按钮</AtButton>)
    component.instance().onClick()
    expect(handleClick).toBeCalled()
  })

  it('handle Button -- when prop disabled, event onClick not to be called', () => {
    const handleClick = jest.fn()
    const component = mount(
      <AtButton disabled onClick={handleClick}>
        按钮
      </AtButton>
    )
    component.instance().onClick()
    expect(handleClick).not.toBeCalled()
  })
})
