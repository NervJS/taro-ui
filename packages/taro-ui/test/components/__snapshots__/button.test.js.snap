// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component render Button -- props circle 1`] = `"<taro-view-core class=\\"at-button at-button--normal at-button--circle\\"><taro-button-core class=\\"at-button__wxbutton\\" lang=\\"en\\"></taro-button-core><taro-view-core class=\\"at-button__text\\">按钮</taro-view-core></taro-view-core>"`;

exports[`Button Component render Button -- props disabled 1`] = `"<taro-view-core class=\\"at-button at-button--normal at-button--disabled\\"><taro-view-core class=\\"at-button__text\\">按钮</taro-view-core></taro-view-core>"`;

exports[`Button Component render Button -- props formType 1`] = `"<taro-view-core class=\\"at-button at-button--normal\\"><taro-button-core class=\\"at-button__wxbutton\\" lang=\\"en\\"></taro-button-core><taro-view-core class=\\"at-button__text\\">按钮</taro-view-core></taro-view-core>"`;

exports[`Button Component render Button -- props full 1`] = `"<taro-view-core class=\\"at-button at-button--normal at-button--full\\"><taro-button-core class=\\"at-button__wxbutton\\" lang=\\"en\\"></taro-button-core><taro-view-core class=\\"at-button__text\\">按钮</taro-view-core></taro-view-core>"`;

exports[`Button Component render Button -- props lang 1`] = `"<taro-view-core class=\\"at-button at-button--normal\\"><taro-button-core class=\\"at-button__wxbutton\\" lang=\\"zh_CN\\"></taro-button-core><taro-view-core class=\\"at-button__text\\">按钮</taro-view-core></taro-view-core>"`;

exports[`Button Component render Button -- props loading 1`] = `"<taro-view-core class=\\"at-button at-button--icon at-button--normal\\"><taro-button-core class=\\"at-button__wxbutton\\" lang=\\"en\\"></taro-button-core><taro-view-core class=\\"at-button__icon\\"><taro-view-core class=\\"at-loading\\"><taro-view-core class=\\"at-loading__ring\\"></taro-view-core><taro-view-core class=\\"at-loading__ring\\"></taro-view-core><taro-view-core class=\\"at-loading__ring\\"></taro-view-core></taro-view-core></taro-view-core><taro-view-core class=\\"at-button__text\\">按钮</taro-view-core></taro-view-core>"`;

exports[`Button Component render Button -- props loading(with primary type) 1`] = `"<taro-view-core class=\\"at-button at-button--icon at-button--normal at-button--primary\\"><taro-button-core class=\\"at-button__wxbutton\\" lang=\\"en\\"></taro-button-core><taro-view-core class=\\"at-button__icon\\"><taro-view-core class=\\"at-loading\\"><taro-view-core class=\\"at-loading__ring\\" style=\\"border: 1px solid #fff; border-color: #fff transparent transparent transparent;\\"></taro-view-core><taro-view-core class=\\"at-loading__ring\\" style=\\"border: 1px solid #fff; border-color: #fff transparent transparent transparent;\\"></taro-view-core><taro-view-core class=\\"at-loading__ring\\" style=\\"border: 1px solid #fff; border-color: #fff transparent transparent transparent;\\"></taro-view-core></taro-view-core></taro-view-core><taro-view-core class=\\"at-button__text\\">按钮</taro-view-core></taro-view-core>"`;

exports[`Button Component render Button -- props loading(with small size) 1`] = `"<taro-view-core class=\\"at-button at-button--icon at-button--small\\"><taro-button-core class=\\"at-button__wxbutton\\" lang=\\"en\\"></taro-button-core><taro-view-core class=\\"at-button__icon\\"><taro-view-core class=\\"at-loading\\"><taro-view-core class=\\"at-loading__ring\\"></taro-view-core><taro-view-core class=\\"at-loading__ring\\"></taro-view-core><taro-view-core class=\\"at-loading__ring\\"></taro-view-core></taro-view-core></taro-view-core><taro-view-core class=\\"at-button__text\\">按钮</taro-view-core></taro-view-core>"`;

exports[`Button Component render Button -- props size(normal) 1`] = `"<taro-view-core class=\\"at-button at-button--normal\\"><taro-button-core class=\\"at-button__wxbutton\\" lang=\\"en\\"></taro-button-core><taro-view-core class=\\"at-button__text\\">按钮</taro-view-core></taro-view-core>"`;

exports[`Button Component render Button -- props size(small) 1`] = `"<taro-view-core class=\\"at-button at-button--small\\"><taro-button-core class=\\"at-button__wxbutton\\" lang=\\"en\\"></taro-button-core><taro-view-core class=\\"at-button__text\\">按钮</taro-view-core></taro-view-core>"`;

exports[`Button Component render Button -- props type(primary) 1`] = `"<taro-view-core class=\\"at-button at-button--normal at-button--primary\\"><taro-button-core class=\\"at-button__wxbutton\\" lang=\\"en\\"></taro-button-core><taro-view-core class=\\"at-button__text\\">按钮</taro-view-core></taro-view-core>"`;

exports[`Button Component render Button -- props type(secondary) 1`] = `"<taro-view-core class=\\"at-button at-button--normal at-button--secondary\\"><taro-button-core class=\\"at-button__wxbutton\\" lang=\\"en\\"></taro-button-core><taro-view-core class=\\"at-button__text\\">按钮</taro-view-core></taro-view-core>"`;

exports[`Button Component render initial Button 1`] = `"<taro-view-core class=\\"at-button at-button--normal\\"><taro-button-core class=\\"at-button__wxbutton\\" lang=\\"en\\"></taro-button-core><taro-view-core class=\\"at-button__text\\">按钮</taro-view-core></taro-view-core>"`;
